Python/pywarpx/picmi.py

```diff
@@ -1908,6 +1908,32 @@ def set_write_dir(self):
             file_prefix = (self.file_prefix or self.name)
             self.diagnostic.file_prefix = os.path.join(write_dir, file_prefix)
 
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class ParticleFieldDiagnostic(object):
```

Comment:
Thanks for adding this - it looks good! Though I have comments.
Here, the `(object)` is not necessary.
```class ParticleFieldDiagnostic:```